packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts

```diff
@@ -133,6 +133,22 @@ const baseCases = [
     code: 'foo.bar && foo.bar?.() && foo.bar?.().baz',
     output: 'foo.bar?.()?.baz',
   },
+  {
+    code: 'foo !== null && foo.bar !== null',
+    output: 'foo?.bar != null',
+  },
+  {
+    code: 'foo != null && foo.bar != null',
+    output: 'foo?.bar != null',
+  },
+  {
+    code: 'foo != null && foo.bar !== null',
+    output: 'foo?.bar != null',
+  },
+  {
+    code: 'foo !== null && foo.bar != null',
+    output: 'foo?.bar != null',
+  },
```

Comment:
🤔 now that #5594 has been merged into this branch, these cases are no longer reporting any errors. What do you see as the right behavior @omril1 @sviat9440?

Reply:
Let me think about it.
My first thought was that we could filter them out, but it might be better to split the base cases into groups

Reply:
Added selectors

Reply:
IMO something like this can be more readable:

```typescript
const baseCases = [/* ... */];
const nonStrictEqualCases = [/* ... */];
const negatedOrCases = baseCases.map(c => ({
  ...c,
  code: c.code.replace(/foo/g, '!foo').replace(/&&/g, '||'),
  errors: [
    {
      ...c.errors[0],
      suggestions: [
        {
          ...c.errors[0].suggestions![0],
          output: `!${c.errors[0].suggestions![0].output}`,
        },
      ],
    },
  ],
}));
const allBasesCases = [...baseCases, ...nonStrictEqualCases];

/* 
  ...later on
	invalid: [
		...allBasesCases.map(...),
		...allBasesCases.map(...),
		...negatedOrCases,
  ]
  */
```

With variable names subject to change

Reply:
I think that in the future additional conditions may appear, and this will increase the number of variables by 2^n.